-D_FORTIFY_SOURCE=2 is enable in gcc by default on alpine. See the
010_all_default-fortify-source.patch patch in main/gcc. Disabling
it here silences a bunch of compiler warnings.

diff -upr vis-0.1.orig/config.mk vis-0.1/config.mk
--- vis-0.1.orig/config.mk	2016-02-27 17:42:14.812754756 +0100
+++ vis-0.1/config.mk	2016-02-27 17:42:23.349390641 +0100
@@ -64,7 +64,7 @@ STRIP ?= strip
 
 # Hardening
 ifeq (${CC},gcc)
-	CFLAGS += -fPIC -fstack-protector-all -D_FORTIFY_SOURCE=2
+	CFLAGS += -fPIC -fstack-protector-all
 	LDFLAGS += -z now -z relro -pie
 else ifeq (${CC},clang)
 	CFLAGS += -fPIE -fstack-protector-all -D_FORTIFY_SOURCE=2
